cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-rococo/src/tests/reserve_transfer.rs

```diff
@@ -469,149 +663,260 @@ fn reserve_transfer_native_asset_from_para_to_system_para() {
-		args: TestArgs::new_para(destination, beneficiary_id, amount_to_send, assets, None, 0),
+		sender: sender.clone(),
+		receiver: receiver.clone(),
+		args: TestArgs::new_para(destination, receiver.clone(), amount_to_send, assets, None, 0),
 	};
+	let mut test = ParaToParaThroughRelayTest::new(test_args);
 
-	let mut test = ParaToParaTest::new(test_args);
+	// Query initial balances
+	let sender_assets_before = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(relay_native_asset_location.clone(), &sender)
+	});
+	let receiver_assets_before = PenpalB::execute_with(|| {
+		type ForeignAssets = <PenpalB as PenpalBPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(relay_native_asset_location.clone(), &receiver)
+	});
 
-	let sender_balance_before = test.sender.balance;
-	let receiver_balance_before = test.receiver.balance;
+	// Set assertions and dispatchables
+	test.set_assertion::<PenpalA>(para_to_para_through_relay_sender_assertions);
+	test.set_assertion::<Rococo>(para_to_para_relay_hop_assertions);
+	test.set_assertion::<PenpalB>(para_to_para_through_relay_receiver_assertions);
+	test.set_dispatchable::<PenpalA>(para_to_para_through_relay_limited_reserve_transfer_assets);
+	test.assert();
 
-	let sender_as_seen_by_relay = Rococo::child_location_of(PenpalA::para_id());
-	let sov_of_sender_on_relay = Rococo::sovereign_account_id_of(sender_as_seen_by_relay);
+	// Calculate delivery fees
+	let delivery_fees = PenpalA::execute_with(|| {
+		xcm_helpers::transfer_assets_delivery_fees::<
+			<PenpalRococoXcmConfig as xcm_executor::Config>::XcmSender,
+		>(test.args.assets.clone(), 0, test.args.weight_limit, test.args.beneficiary, test.args.dest)
+	});
+
+	// Query final balances
+	let sender_assets_after = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(relay_native_asset_location.clone(), &sender)
+	});
+	let receiver_assets_after = PenpalB::execute_with(|| {
+		type ForeignAssets = <PenpalB as PenpalBPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(relay_native_asset_location.clone(), &receiver)
+	});
+
+	// Sender's balance is reduced
+	assert_eq!(sender_assets_before - amount_to_send - delivery_fees, sender_assets_after);
+	// Receiver's balance is increased
+	assert!(receiver_assets_after > receiver_assets_before);
+}
+
+/// Reserve Transfers of native asset from Parachain to Parachain (through Relay reserve) should
+/// work
+#[test]
+fn reserve_transfer_native_asset_from_para_to_para_through_system_para() {
+	// Init values for Parachain Origin
+	let destination = PenpalA::sibling_location_of(PenpalB::para_id());
+	let sender = PenpalASender::get();
+	let amount_to_send: Balance = ASSET_HUB_ROCOCO_ED * 10000;
+	let asset_owner = penpal_asset_owner();
+	let assets: Assets = ((Parent, Parachain(ASSETHUB_PARA_ID)), amount_to_send).into();
```

Comment:
What is this asset?

IIUC this PR makes this is sort of a "symlink" to `RelayNative`. Aka you propose that all parachains be able to interchangeably say `(1, Parachain(ASSETHUB_PARA_ID))` and `(1, Here)` while still referring to DOT/KSM.

I don't see anything wrong with this on the surface, but it is a big conceptual change that the whole ecosystem needs to be aware of (and align to). In my opinion this definitely needs a [Polkadot RFC](https://github.com/polkadot-fellows/RFCs) before implementing/merging.